### DISCUSSION TITLE
Extension: Add a new Wiki Table extension

### DISCUSSION THREAD

**Discussion by movsb** [yuin/goldmark#428] [Ideas]
Hi, 😊

I just found that [Wiki Table](https://www.mediawiki.org/wiki/Help:Tables) can have complex/beautiful Tables in simple texts. So I wrote an extension that can parse paragraphs containing Wiki Table and transform it into an HTML Table. Here is the source code link: <https://github.com/movsb/goldmark-wiki-table>. And you can try it here: <https://movsb.github.io/goldmark-wiki-table/>.

**Comment by yuin** [comment]
Good extension. You can create a new pull request like #427.

**Comment by movsb** [comment]
#429

**Reply by tomcam** [reply]
Fantastic extension. Can you help me with something? Looking at the source for your demo at [https://movsb.github.io/goldmark-wiki-table/](https://movsb.github.io/goldmark-wiki-table/) I see a parse() function. Where does that come from?

**Reply by movsb** [reply]
Hi, the `parse` function is from Go, compiled to WebAssembly. Check it [here](https://github.com/movsb/goldmark-wiki-table/blob/b86e7746f557207defeeded9b2642c40278bb993/docs/worker.go#L25).

**Comment by tomcam** [comment]
Thank you! Somehow I thought this was forcing the HYML render

**Reply by movsb** [reply]
Oh sorry, yes, I hardcoded that while extending the Markdown instance for simplicity.  Think that this is done by most of the extensions. Their implementation code, however, is separate and can be used independently.
